spec/02-integration/05-proxy/10-balancer/01-healthchecks_spec.lua

```diff
@@ -217,7 +221,13 @@ for _, strategy in helpers.each_strategy() do
 
       -- we do not set up servers, since we want the connection to get refused
       -- Go hit the api with requests, 1x round the balancer
-      local oks, fails, last_status = bu.client_requests(bu.SLOTS, api_host)
+      local oks, fails, last_status
+      helpers.wait_until(function()
+        oks, fails, last_status = bu.client_requests(bu.SLOTS, api_host)
+        return last_status == 503
```

Comment:
is this new / changed behaviour or was this always the case and we just never tested?

Reply:
The new event mechanism will not be blocked to sync, which means it is async. so we have to wait until it succeeds.